### ISSUE TITLE
SourceGenerator fails on interfaces with the same name

### ISSUE THREAD

**Issue by dtewinkel** [canton7/RestEase#252]
**Description**

Source generation fails on interfaces with the same name, but in different namespaces. 
The warning during compilation is: `CSC : warning CS8785: Generator 'RestEaseSourceGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentException' with message 'The hintName 'RestEase_ISomeApi.g.cs' of the added source file must be unique within a generator.`. The resulting application still works, but without the generated sources and thus dynamically generating the implementations at run-time. 

**To Reproduce**

Using the assemblies:

- RestEase
- RestEase.SourceGenerator

Create the following application and compile it:

```cs
using System.Threading.Tasks;
using RestEase;

namespace SourceGeneratorSandbox
{
    public record HttpBinResponse
    {
        public string? Origin { get; set; }

        public string? Url { get; set; }
    }

    public class Program
    {
        public static async Task Main()
        {
            var impl = RestClient.For<ISomeApi>("https://httpbin.org");
            var response = await impl.GetAsync();

            var impl2 = RestClient.For<AnotherNamespace.ISomeApi>("https://httpbin.org");
            var response2 = await impl2.PostAsync();
        }

        public interface ISomeApi
        {
            [Get("/get")]
            Task<HttpBinResponse> GetAsync();
        }
    }

    namespace AnotherNamespace
    {
        public interface ISomeApi
        {
            [Post("/post")]
            Task<HttpBinResponse> PostAsync();
        }
    }
}
```

**Version Info**

 - RestEase version: 1.6.1
 - Target framework version: net6.0

**Additional Info**

**Comment by canton7** [comment]
Apologies for the delay -- I was away. Fixed in 1.6.2.

**Comment by dtewinkel** [comment]
No problem. Thank you 😄
